src/properties/rich_club.c

```diff
@@ -0,0 +1,167 @@
+ * lengths
+ *
+ * Time complexity: O(|V| + |E|)
+ * where |V| is the number of vertices and |E| the number of edges in the graph given.
+ *
+ * \sa \ref igraph_density(), which uses the same calculation of total possible edges.
+ */
+igraph_error_t igraph_rich_club_density_sequence(const igraph_t *graph,
+                                                 const igraph_vector_int_t *vertex_order,
+                                                 igraph_bool_t directed,
+                                                 igraph_bool_t loops,
+                                                 igraph_vector_t *weights,
+                                                 igraph_vector_t *res) {
+
+    const igraph_integer_t numVertices = igraph_vcount(graph);
+    const igraph_integer_t numEdges = igraph_ecount(graph);
+
+    // Error handling: invalid vertex_order and weights sizes
+    if (igraph_vector_int_size(vertex_order) != numVertices) {
+        IGRAPH_ERRORF("Vertex order vector length (%" IGRAPH_PRId ") does not match "
+                      "number of vertices (%" IGRAPH_PRId ").",
+                      IGRAPH_EINVAL,
+                      igraph_vector_int_size(vertex_order), numVertices);
+    }
+    if (weights && igraph_vector_size(weights) != numEdges) {
+        IGRAPH_ERRORF("Weight vector length (%" IGRAPH_PRId ") does not match "
+                      "number of edges (%" IGRAPH_PRId ").",
+                      IGRAPH_EINVAL,
+                      igraph_vector_size(weights), numEdges);
+    }
+
+    igraph_vector_t edgesRemainingAfter;
+    igraph_vector_int_t orderOf;
+
+    IGRAPH_VECTOR_INIT_FINALLY(&edgesRemainingAfter, numVertices);
+    IGRAPH_VECTOR_INT_INIT_FINALLY(&orderOf, numVertices);
+
+    IGRAPH_CHECK(igraph_vector_resize(res, numVertices)); // resize res
+    IGRAPH_CHECK(get_vertex_order_map(vertex_order, &orderOf)); // get map of vertex order
+
+    igraph_bool_t warningIssued = false;
+
+    // edgesRemainingAfter vector: number of edges (or total edge weight) removed by index
+    for (igraph_integer_t eid = 0; eid < numEdges; eid++) {
+        igraph_integer_t v1 = IGRAPH_FROM(graph, eid);
+        igraph_integer_t v2 = IGRAPH_TO(graph, eid);
+        if (v1 == v2 && !loops && !warningIssued) {
+            IGRAPH_WARNING("Self-loop encountered while `loops = false`. "
+                           "Proceeding as if loops were not possible (density computed accordingly)");
+            warningIssued = true;
+        }
+        igraph_integer_t orderV1 = VECTOR(orderOf)[v1]; // order of endpoints
+        igraph_integer_t orderV2 = VECTOR(orderOf)[v2];
+
+        igraph_integer_t edgeRemovalIndex = (orderV1 < orderV2 ? orderV1 : orderV2);
+        VECTOR(edgesRemainingAfter)[edgeRemovalIndex] += (weights ? VECTOR(*weights)[eid] : 1);
+    }
+
+    // edgesRemainingAfter vector: edges (or total edge weight) remaining after i removals
+    igraph_integer_t total = 0;
```

Comment:
Attention, now we have reals since we're dealing with weights. This is a common mistake when generalizing a non-weighted implementation to a weighted one. I'll commit the fix, you don't need t worry about it now.

```suggestion
    igraph_integer_t total = 0;
```